### PR TITLE
Batch Tracer Endpoint & Tracer Sampling

### PR DESCRIPTION
**Pull request by hinthornw**

### CODE REVIEW

File: python/langsmith/client.py [hunk 1]
```diff
@@ -690,13 +715,64 @@ def upload_csv(
             **result, _host_url=self._host_url, _tenant_id=self._get_tenant_id()
         )
 
+    def _run_transform(
```
Comment (jacoblee93) [inline python/langsmith/client.py:718]: How about calling this `_prepare_run_create_params()`?

Comment (jacoblee93) [reply python/langsmith/client.py:718]: Also, does it need to be an instance method?

File: python/langsmith/client.py [hunk 2]
```diff
@@ -690,13 +715,64 @@ def upload_csv(
...
+            run_create = cast(dict, run)
+        if isinstance(run["id"], str):
+            run["id"] = uuid.UUID(run["id"])
+        if "inputs" in run_create:
+            run_create["inputs"] = _hide_inputs(run_create["inputs"])
+        if "outputs" in run_create:
+            run_create["outputs"] = _hide_outputs(run_create["outputs"])
+        return run_create
+
+    def _insert_runtime_env(self, runs: Sequence[dict]) -> None:
```
Comment (jacoblee93) [inline python/langsmith/client.py:733]: Same as above, does this need to be an instance method? And naming could be clearer

File: python/langsmith/client.py [hunk 3]
```diff
@@ -690,13 +715,64 @@ def upload_csv(
...
 
+    def _run_transform(
+        self, run: Union[ls_schemas.Run, dict, ls_schemas.RunLikeDict]
+    ) -> dict:
+        if hasattr(run, "dict") and callable(getattr(run, "dict")):
+            run_create = run.dict()  # type: ignore
+        else:
+            run_create = cast(dict, run)
+        if isinstance(run["id"], str):
+            run["id"] = uuid.UUID(run["id"])
```
Comment (jacoblee93) [inline python/langsmith/client.py:726]: Shouldn't this be `run_create`?

File: python/langsmith/client.py (hunk 2 above)
Comment (jacoblee93) [reply python/langsmith/client.py:733]: `_merge_runtime_env_into_create_run_param_list`?

File: python/langsmith/client.py (hunk 1 above)
Comment (hinthornw) [reply python/langsmith/client.py:767]: No it doesn't

File: python/langsmith/client.py (hunk 3 above)
Comment (hinthornw) [reply python/langsmith/client.py:777]: Or update

**Pull request merged**